### PR TITLE
feat(org): wire org code contributions page to warehouse data

### PR DESCRIPTION
**Pull request by ahmedomosanya**
## Summary
- Delete `org-contributions.demo.ts`; `OrgContributionsService` now reads `platinum_lfx_one_org_code_contributions` via Snowflake for KPIs, repositories, the commits feed, and the project/employee filter options.
- Add a server-side **Commits** view alongside Repositories: tab-aware KPI search, dedicated `commitSort`/`commitDir`, independent pagination, and `commitsTotalRecords` for the tab count + pager.
- Thread contributor identity and avatar through the commits feed (`contributorId`, `committerAvatarUrl`) so panel grouping is stable and initials are only a fallback.
- Validate the new `view` and `commitSort` query params at the controller boundary; pass the request through to the service.
- Carry org `uid`/`parentUid` through account context and persona account seeding so the page resolves the selected org.
- https://linuxfoundation.atlassian.net/browse/LFXV2-1894

[LFXV2-1894]: https://linuxfoundation.atlassian.net/browse/LFXV2-1894?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts [hunk 1]
```diff
@@ -324,11 +313,11 @@ export class OrgContributionsComponent {
...
 
   private initCommitterDetail(): OrgCommitterDetailVm | null {
-    const name = this.selectedCommitter();
-    if (!name) {
+    const contributorId = this.selectedContributorId();
+    if (!contributorId) {
       return null;
     }
-    const rows = this.commitRows().filter((c) => c.committerName === name);
+    const rows = this.commitRows().filter((c) => c.contributorId === contributorId);
```
[the remark this answers is not in the document]
Comment (ahmedomosanya) [reply apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts]: Addressed in `424e5f07` — committer name renders as plain text when `contributorId` is null; the panel opens only when a stable member id is present (no `commitSha` fallback).

File: apps/lfx-one/src/server/services/org-contributions.service.ts [hunk 2]
```diff
@@ -2,200 +2,480 @@
...
+  return {
+    commitSha: row.COMMIT_ID,
+    contributorId: row.MEMBER_ID,
+    projectName: row.PROJECT_NAME ?? '—',
+    committerName: row.MEMBER_DISPLAY_NAME ?? 'Unknown',
+    committerAvatarUrl: row.MEMBER_LOGO,
+    committerTitle: null,
+    username: parseGithubUsername(row.GITHUB_USERNAME),
+    source: row.SOURCE ?? 'git',
+    committedTs: toIsoTimestamp(row.ACTIVITY_TS) ?? new Date(0).toISOString(),
```
[the remark this answers is not in the document]
Comment (ahmedomosanya) [reply apps/lfx-one/src/server/services/org-contributions.service.ts]: Addressed in `424e5f07` — null `ACTIVITY_TS` maps to an empty `committedTs`; `decorateCommitFeedRow` already renders `—` for falsy timestamps.

File: apps/lfx-one/src/server/services/org-contributions.service.ts [hunk 3]
```diff
@@ -2,200 +2,480 @@
...
+    return '';
+  }
+  const placeholders = projects.map(() => '?').join(', ');
+  for (const slug of projects) {
+    binds.push(slug);
+  }
+  for (const slug of projects) {
+    binds.push(slug);
+  }
+  return `AND (project_slug IN (${placeholders}) OR parent_slug IN (${placeholders}))`;
```
[the remark this answers is not in the document]
Comment (ahmedomosanya) [reply apps/lfx-one/src/server/services/org-contributions.service.ts]: Addressed in `424e5f07` — `projectFilterPredicate` now matches `project_id IN (...)` alongside `project_slug` and `parent_slug`, so options whose slug falls back to `PROJECT_ID` filter correctly.

File: apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts [hunk 4]
```diff
@@ -324,11 +313,11 @@ export class OrgContributionsComponent {
...
     if (rows.length === 0) {
```
[the remark this answers is not in the document]
Comment (ahmedomosanya) [reply apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts:327]: Addressed in `424e5f07` — removed the `commitSha` fallback from the click path instead of widening the filter. Commits without `contributorId` render the committer name as plain text (not clickable); the panel opens only when a stable member id is present, which matches `initCommitterDetail`'s `contributorId` filter.

File: apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.html [hunk 5]
```diff
@@ -325,7 +352,7 @@ <h1 class="text-2xl font-semibold text-gray-900" data-testid="org-contributions-
 
   <!-- Committer detail side panel -->
   <p-drawer
-    [visible]="committerDetail() !== null"
+    [visible]="selectedContributorId() !== null"
     (visibleChange)="onCommitterPanelVisible($event)"
```
[the remark this answers is not in the document]
Comment (ahmedomosanya) [reply apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.html:356]: Addressed in 79b6a3c7 — p-drawer `visible` now keys off `committerDetail()` so the panel auto-closes when filter/search/page changes drop the selected contributor from `commitRows()`.

File: apps/lfx-one/src/server/services/org-contributions.service.ts [hunk 6]
```diff
@@ -2,200 +2,483 @@
...
+        employeeOptions,
+        totalRecords,
+        commitsTotalRecords,
+      };
+    } catch (error) {
+      logger.warning(req, 'get_org_lens_contributions', 'Snowflake query failed, returning empty contributions', {
+        error: error instanceof Error ? error.message : String(error),
+        account_id: accountId,
+      });
+      return emptyResponse(accountId, query.dateRange);
```
Comment (audigregorie) [inline apps/lfx-one/src/server/services/org-contributions.service.ts]: **Major — Snowflake failures are swallowed into an empty 200, making the error/retry UI unreachable**

This `catch` downgrades every query failure to an empty `200` response. But the component's error state (`org-contributions.component.html:75` → `@if (fetchError())` with a `retry()` CTA) is only set by the HTTP `catchError` in `initResponse()`, which fires on a non-2xx status. Result: a real warehouse outage renders as "0 commits / empty table" — indistinguishable from a genuinely empty org — and the retry button never shows.

Either rethrow so the controller's `next(error)` surfaces a 5xx (and the existing error/retry UI works), or, if graceful-empty is the intended product behavior, remove the now-dead `fetchError`/`retry` affordance and document it. As written, the two halves contradict each other.

File: apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts (hunk 1 above)
Comment (audigregorie) [inline apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts]: **Major — committer panel now aggregates only the current page of commits**

This PR makes the Commits feed server-paginated (the commits table gains `[lazy]` + `[paginator]`), so `commitRows()` is now a single page (≤ `size`). But `initCommitterDetail()` still derives the panel entirely from `commitRows()`:

- `totalCommits: rows.length` → counts only this contributor's commits **on the current page**
- `projects` and the Code-tab `commits` list → likewise page-scoped

Pre-PR the feed was unpaginated client-side, so the panel saw the whole filtered set. Now a contributor with 200 commits shows "10 commits" when opened from page 1. Consider fetching the contributor's full commit set for the panel (a query keyed by `contributorId`), or relabel the count/list as page-scoped so it isn't presented as a total.

File: apps/lfx-one/src/server/services/org-contributions.service.ts [hunk 7]
```diff
@@ -2,200 +2,483 @@
...
+  const placeholders = projects.map(() => '?').join(', ');
+  for (const slug of projects) {
+    binds.push(slug);
+  }
+  for (const slug of projects) {
+    binds.push(slug);
+  }
+  for (const slug of projects) {
+    binds.push(slug);
+  }
```
Comment (audigregorie) [inline apps/lfx-one/src/server/services/org-contributions.service.ts]: **Minor — collapse the three identical bind loops**

The three IN-lists each need the same binds in the same order; one loop reads cleaner.

```suggestion
  // project_slug / parent_slug / project_id each get the same binds, in order.
  for (let i = 0; i < 3; i += 1) {
    binds.push(...projects);
  }
```

**Review by audigregorie — COMMENTED** [review]
## Code Review Summary

Solid warehouse wiring — parameterized SQL, clean controller validation, good null handling. Two behavioral concerns worth addressing before merge (error swallowing + page-scoped committer panel), plus one readability nit. Posting findings as individual threads.

### Outside the diff — question (low confidence)

`fetchRepositories` computes `first_commit_ts` as `MIN(org_repo_first_commit_ts)` over rows already filtered by `activity_ts` within the date window. When a non-`all` range is active, this returns the earliest *first-commit* among repos with activity in the window — but could it exclude a repo whose `org_repo_first_commit_ts` predates the window? If `org_repo_first_commit_ts` is a precomputed repo-level constant this is fine; flagging only to confirm the warehouse semantics are intended.

### Note — overall design (perf, optional)

`getContributions` always runs both `fetchRepositories` and `fetchCommits` (full page rows) every request, even for the inactive tab. The inactive tab only needs its `COUNT(*) OVER()` for the badge, not a page of rows. Fetching up to `size` rows (incl. `commit_message`) for the hidden table is minor wasted warehouse work — acceptable as a tradeoff, noting for awareness.

### What's done well

- SQL fully parameterized; `ORDER BY`/dir and `LIMIT/OFFSET` driven only by enum/numeric-validated values — no injection surface.
- Clean controller-boundary validation for `view` / `commitSort` (400 on bad input, defaults on missing).
- `toIsoTimestamp` robustly handles `Date | string | null` and null paths render as `—`.

File: apps/lfx-one/src/server/services/org-contributions.service.ts (hunk 6 above)
Comment (ahmedomosanya) [reply apps/lfx-one/src/server/services/org-contributions.service.ts]: Addressed in `90dc40d6` — removed the service-level catch that returned `emptyResponse` on Snowflake failure. Query errors now propagate to the controller `next(error)` path so the existing `fetchError` / `retry()` UI surfaces on warehouse outages.

File: apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts (hunk 1 above)
Comment (ahmedomosanya) [reply apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts]: Addressed in `90dc40d6` — relabeled the panel project/commit sections as page-scoped ("Projects in view", "Recent commits in view"; commit count already read "Commits in view"). Full contributor-scoped fetch for the drawer is deferred to LFXV2-2195 (shared person detail drawer).

File: apps/lfx-one/src/server/services/org-contributions.service.ts (hunk 7 above)
Comment (ahmedomosanya) [reply apps/lfx-one/src/server/services/org-contributions.service.ts]: Addressed in `90dc40d6` — collapsed the three identical `project_slug` / `parent_slug` / `project_id` bind loops into a single `for (let i = 0; i < 3; i += 1) binds.push(...projects)` pass.

File: apps/lfx-one/src/server/services/org-contributions.service.ts [hunk 8]
```diff
@@ -2,200 +2,468 @@
...
+    const repoSearch = buildRepoSearchFilter(query.search);
+    const commitSearch = buildCommitSearchFilter(query.search);
+
+    const [kpiResult, repoResult, commitResult, projectOptions, employeeOptions] = await Promise.all([
+      this.fetchKpis(accountId, scope, kpiSearch),
+      this.fetchRepositories(accountId, scope, repoSearch, query),
+      this.fetchCommits(accountId, scope, commitSearch, query),
+      this.fetchProjectOptions(accountId, query.dateRange),
+      this.fetchEmployeeOptions(accountId, query.dateRange),
+    ]);
```
[the remark this answers is not in the document]
Comment (ahmedomosanya) [reply apps/lfx-one/src/server/services/org-contributions.service.ts:99]: Addressed in `092f45cc` — `viewAwarePagination` applies `page`/`size` only to `query.view`; the inactive tab fetches page 1 / size 1 to read `COUNT(*) OVER()` totals without returning rows, so tab counts stay correct on out-of-range pages.

File: apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts (hunk 1 above)
Comment (MRashad26) [inline apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts]: 🟡 **Warning — committer detail panel now sees only the current page of commits, causing under-counted `totalCommits` and missing commits in the panel.**

`this.commitRows()` is now the server-paginated current page (default 10 rows). A contributor who appears on page 1 may have commits on pages 2 and 3 that the panel never shows, and `totalCommits: rows.length` (line 337) reports only the on-page count.

Before this PR, all commits were loaded client-side so the panel always showed a contributor's full commit history. This is a functional regression introduced by moving to server-side pagination.

**Suggested fix options:**
1. When `openCommitter(contributorId)` is called, fire a dedicated BFF request `GET /api/org-contributions/:orgId/committer/:contributorId` that fetches all commits for that contributor (no pagination) and populates the panel from that response.
2. Pass `contributorId` as a filter to the existing endpoint with a large `size` (e.g. 200) and no pagination UI on the panel — simpler but caps the history.

Option 1 is cleaner long-term; option 2 unblocks the immediate regression with minimal change.

File: apps/lfx-one/src/server/services/org-contributions.service.ts [hunk 9]
```diff
@@ -2,200 +2,487 @@
...
+function viewAwarePagination(query: OrgContributionsQuery, tab: ContributionsView): ViewAwarePagination {
+  if (query.view === tab) {
+    return { page: query.page, size: query.size };
   }
-  const names = new Set<string>();
-  for (const option of DEMO_PROJECT_OPTIONS) {
-    if (slugs.has(option.slug)) {
-      names.add(option.name);
-    }
+  return { page: 1, size: 1 };
```
Comment (MRashad26) [inline apps/lfx-one/src/server/services/org-contributions.service.ts:309]: 🔴 **Critical — 5 of 6 commits on this PR carry Claude co-author trailers, violating CLAUDE.md.**

Commit list:
```
8a49d1b88  message body: "Generated with [Cursor Composer]"
            trailer:      Co-Authored-By: Claude Opus 4.8 <noreply@anthropic.com>
424e5f078  Co-Authored-By: Claude Opus 4.8 <noreply@anthropic.com>
79b6a3c7a  Co-Authored-By: Claude Opus 4.7 <noreply@anthropic.com>
90dc40d6c  Co-Authored-By: Claude Opus 4.8 <noreply@anthropic.com>
092f45cc9  Co-Authored-By: Claude Opus 4.7 <noreply@anthropic.com>
```
Only `31e16fe9b` is clean.

**Fix:** interactive rebase to remove all `Co-Authored-By` trailers and the `Generated with [Cursor Composer]` line from `8a49d1b88`'s message body before merging.

_(Anchoring to the `viewAwarePagination` function as the nearest changed line in this file; the violation is in the commit metadata.)_

**Review by MRashad26 — COMMENTED** [review]
Automated review + `code-standards-enforcer` audit (report-only). 6 commits; 5 signed off by Michal Lehotsky. **5 of 6 commits carry Claude co-author trailers — see Critical section.**

## 🔒 Secrets / critical-constants check
No secrets or credentials. `PLATINUM_TABLE` is a hardcoded constant. All user-supplied filter values go through `?` placeholders; the only template-interpolated SQL fragments (`dateRangePredicate`, `orderBy`, `sortDir`) derive exclusively from validated enum values — no injection surface. ✅

## Overview
Replaces the LFXV2-1894 demo scaffold with live Snowflake warehouse reads. The BFF service now runs five parallel queries per request (KPIs, repos, commits, project options, employee options) against `ANALYTICS.PLATINUM_LFX_ONE.ORG_CODE_CONTRIBUTIONS`. Key improvements:
- `view` (active tab) propagated end-to-end: URL → Angular → BFF → SQL
- Commits tab is now server-paginated and server-sorted (was client-side sort+slice)
- `contributorId` (Crowd.dev `member_id`) replaces committer name as the panel grouping key — correct fix for name-collision risk
- Avatar images via `member_logo` with initials fallback
- `viewAwarePagination` fires both queries on every request but reduces the inactive-tab query to `LIMIT 1` to read `COUNT(*) OVER()` cheaply
- `account-context.service.ts` preserves `uid`/`parentUid` from the initial seed when the live account map doesn't carry them
- Demo file deleted cleanly; `assertDemoAllowed()` guard removed

## Correctness

**🔴 Critical — commit process violations (see inline)**

**🟡 Warning — committer detail panel sees only the current page of commits**
`initCommitterDetail()` (line 315) filters `this.commitRows()` to build the side-panel rows. Before this PR, all commits were loaded client-side, so the panel always showed every commit for the selected contributor. Now `commitRows()` is the server-paginated current page (up to `size` rows, default 10). A contributor who has commits across multiple pages will appear clickable on page 1, but the panel will only show the commits on that page and `totalCommits: rows.length` (line 337) will under-count. This is a functional regression hidden by the architecture change.

**🔵 Info — double Snowflake queries on every BFF request**
`viewAwarePagination` correctly limits the inactive tab to `LIMIT 1 OFFSET 0`, but both Snowflake queries still execute on every call. For orgs with large datasets both CTEs scan the full `PLATINUM_TABLE` partition regardless of which tab is active. No correctness issue; worth profiling under load and considering a view-conditional short-circuit if query times are high.

**🔵 Info — `QUALIFY ROW_NUMBER()` is Snowflake-specific syntax**
Line 217 of the service uses `QUALIFY ROW_NUMBER() OVER (PARTITION BY commit_id ...) = 1` to deduplicate. This is correct Snowflake SQL and eliminates fan-out rows for the same commit. Just noting it's non-portable if the query layer is ever swapped — as-is it's the idiomatic Snowflake pattern.

**✅ Verified correct:**
- All user-supplied filter values use `?` bind params; `orderBy` and `sortDir` are derived from validated enums
- `toIsoTimestamp` handles both `Date` objects and Snowflake-style `'YYYY-MM-DD HH:MM:SS'` strings; NaN guard returns null
- `parseGithubUsername` handles full profile URLs, `@handle`, and bare handles; regex anchored correctly
- `projectFilterPredicate` pushes project binds three times (slug / parent_slug / project_id) — matches the three `IN (?)` clauses
- `buildKpiSearchFilter` applies the active tab's search semantics to KPI counts — intentional behavior
- Tab switch resets `page` to 1 in `onMainTabChange` and in the `combineLatest` sort observer
- `emptyResponse` short-circuits for empty `accountId` — prevents a query with `account_id = ''`

## Code Standards Audit (`~/LFX/code-enforcer-agent.md`)

**🔴 Critical:**
- **Claude co-author trailers in 5 of 6 commits** (see inline at head file for anchor):
  - `8a49d1b88` — commit *message body* contains `Generated with [Cursor Composer]` AND trailer `Co-Authored-By: Claude Opus 4.8`
  - `424e5f078` — `Co-Authored-By: Claude Opus 4.8`
  - `79b6a3c7a` — `Co-Authored-By: Claude Opus 4.7`
  - `90dc40d6c` — `Co-Authored-By: Claude Opus 4.8`
  - `092f45cc9` — `Co-Authored-By: Claude Opus 4.7`
  - Only `31e16fe9b` is clean (signed-off only).
  Strip all trailers and the Cursor Composer line via interactive rebase before merging.

**🔵 Info:**
- `org-contributions.interface.ts` lines added with inline JSDoc comments on `contributorId` and `committerAvatarUrl` fields — acceptable single-line `/** */` doc-comments on exported interface fields, not multi-line blocks.

**✅ Correctly followed:** demo scaffold deleted (no stale `TODO(LFXV2-1894)` comments remain in prod path), `@lfx-one/shared/interfaces` and `@lfx-one/shared/constants` for all shared types/constants, signals and `computed()` in templates, no functions in template expressions, standalone components, no `CommonModule`, MIT headers present, kebab-case filenames, no nested ternaries, signed-off commits on all 6.

## Test coverage
E2E spec updated: new helpers (`seedSelectedOrgCookie`, `waitForContributionsLoaded`, `switchToCommitsTab`), API stub for `org-lens-account-context`, avatar + pagination assertions. No unit tests for the new SQL-building helpers (`buildScopeFilters`, `projectFilterPredicate`, `toIsoTimestamp`, `parseGithubUsername`) — these are good unit-test targets given their edge-case logic.

## Verdict: **FAIL** (process) / **PASS** (code)
The Snowflake wiring is solid and the data-layer design is correct. Two items before merging: (1) strip the 5 Claude co-author trailers, (2) address the committer-panel pagination regression so the side panel fetches all commits for the selected contributor rather than just the current page.

File: apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts (hunk 1 above)
Comment (ahmedomosanya) [reply apps/lfx-one/src/app/modules/dashboards/org/org-contributions/org-contributions.component.ts]: Addressed in `b0ee1d0f` — opening the committer panel fires a contributor-scoped GET (`employees=[contributorId]`, `view=commits`, `size=100`) that respects the active date/project/search filters; `totalCommits` uses `commitsTotalRecords` from that response, not the paginated table page.

File: apps/lfx-one/src/server/services/org-contributions.service.ts (hunk 9 above)
Comment (ahmedomosanya) [reply apps/lfx-one/src/server/services/org-contributions.service.ts:309]: Addressed — rebased onto `origin/main` and stripped all `Co-Authored-By` / `Generated with [Cursor Composer]` lines from the six feature commits (`0b37a0fa`…`b0ee1d0f`). Verified locally: `git log origin/main..HEAD` has no co-author trailers.

File: apps/lfx-one/src/server/services/org-contributions.service.ts [hunk 10]
```diff
@@ -2,200 +2,487 @@
...
-function expandProjectSelection(selected: string[]): Set<string> | null {
-  if (!selected.length) {
-    return null;
+  private async fetchKpis(accountId: string, scope: ScopeFilters, kpiSearch: SearchFilter): Promise<{ rows: ContributionsKpiRow[] }> {
+    const sql = `
+      SELECT
+        COUNT(DISTINCT project_id) AS PROJECTS_WITH_ACTIVITY,
+        COUNT(DISTINCT repository_url) AS REPOSITORIES,
+        COUNT(DISTINCT commit_id) AS COMMITS
+      FROM ${PLATINUM_TABLE}
```
[the remark this answers is not in the document]
Comment (ahmedomosanya) [reply apps/lfx-one/src/server/services/org-contributions.service.ts:126]: Addressed in `0d455d19` — `projectsWithActivity` now uses `COUNT(DISTINCT COALESCE(project_id, project_slug, project_name))` so rows with a null `project_id` but a slug/name still count toward the KPI.

**Review by audigregorie — APPROVED** [review]
